### PR TITLE
detect-itype.c: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR #8055
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4049) ticket:

Describe changes:

- Update  `DetectITypeParseTest08` in detect-itype unit tests 

suricata-verify-pr #970

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
All tests looking good! :)

One small thing went through: could you fix the commit message? The file extension isn't required in it.

**Comment by hsadia538** [comment]
Should I remove the unit test for `DetectITypeMatchTest01` as I opened a suricata-verify PR for the same in the same PR with another commit or 
Should I open a separate PR with one commit for unit test removal?

**Comment by jufajardini** [comment]
> Should I remove the unit test for `DetectITypeMatchTest01` as I opened a suricata-verify PR for the same in the same PR with another commit or Should I open a separate PR with one commit for unit test removal?

Let's keep that work for a separate PR, as it may take longer to merge the SV test, and this would prevent the merge of this commit, even though it may be ready to go ;)

**Review by jufajardini — APPROVED** [review]
Since the commit message has been updated, the PR LGTM :)

**Comment by hsadia538** [comment]
> > Should I remove the unit test for `DetectITypeMatchTest01` as I opened a suricata-verify PR for the same in the same PR with another commit or Should I open a separate PR with one commit for unit test removal?
> 
> Let's keep that work for a separate PR, as it may take longer to merge the SV test, and this would prevent the merge of this commit, even though it may be ready to go ;)

got it now! Thanks

File: src/detect-itype.c [hunk 1]
```diff
@@ -221,14 +221,13 @@ static bool PrefilterITypeIsPrefilterable(const Signature *s)
 static int DetectITypeParseTest01(void)
 {
     DetectU8Data *itd = NULL;
-    int result = 0;
     itd = DetectITypeParse(NULL, "8");
```
Comment (victorjulien) [inline src/detect-itype.c:224]: no need to update this PR, but in future PRs can you update such tests to declare `itd` on the line where it is set? So:
```c
DetectU8Data *itd = DetectITypeParse(...
```

**Review by jasonish — APPROVED** [review]
Looks good to me.

**Comment by victorjulien** [comment]
Merged in #8113, thanks!

**Pull request closed without merging**